test/unit/test_compression_util.py

```diff
@@ -26,3 +39,13 @@ def assert_safety(self, path, expected_to_be_safe):
                     CompressedFile(path).extract(temp_dir)
         finally:
             shutil.rmtree(temp_dir, ignore_errors=True)
+
+    def assert_format_detected(self, path, expected_fmt, allowed_fmts=None):
+        fmt1 = get_fileobj_raw(path, "r", allowed_fmts)[0]
```

Comment:
I'd do this in a for loop and also verify that we get the correct type:
```
for mode, expected_type in [('r', six.text_type), ('rb', six.binary_type), ('rt', six.text_type), ('U', six.text_type)]:
    fmt, fh = get_fileobj_raw(path, mode, allowed_fmts)
    assert fmt == expected_fmt
    assert isinstance(fh.read(1), expected_type)
```
I also like to avoid unittest's assertEqual (or unittest functionality in general), pytest has nice reporting for assertion errors.

Reply:
Thanks for the nice idea of using the six types!
The only complication I had to deal with on top is that when the file format does not get autodetected, then, in text mode, the stream might not be decodable using utf-8. I decided to just skip the type check in this situation.

Reply:
Ha, found a more elegant approach. It is possible to read 0 bytes/characters from the stream to learn whether it's binary or text. This avoids the encoding issue.